spec/classes/init_spec.rb

```diff
@@ -1,439 +1,530 @@
     end
   end
 
   describe 'running on unsupported OS' do
-    let :facts do
-      {
-        :osfamily => 'WierdOS',
-        :group    => 'foo',
-      }
-    end
+    let(:facts) { { osfamily: 'WierdOS' } }
 
-    it 'should fail' do
+    it 'fails' do
       expect {
-        should contain_class(subject)
-      }.to raise_error(Puppet::Error,/Operating system family WierdOS is not supported/)
+        is_expected.to contain_class(:subject)
+      }.to raise_error(Puppet::Error, %r{Operating system family WierdOS is not supported})
     end
   end
 
   describe 'variable type and content validations' do
     # set needed custom facts and variables
-    let(:facts) { {
-      :osfamily => 'RedHat',
-      :fqdn     => 'hieramerge.example.local',
-      :group    => 'spectest',
-    } }
-    let(:validation_params) { {
-#      :param => 'value',
-    } }
+    let(:facts) { { group: 'data_from_hiera_group' } }
 
     validations = {
-# use validate_absolute_path()
-#      'absolute_path' => {
-#        :name    => ['autofs_sysconfig','autofs_auto_master'],
-#        :valid   => ['/absolute/filepath','/absolute/directory/'],
-#        :invalid => ['invalid',3,2.42,['array'],a={'ha'=>'sh'}],
-#        :message => 'is not an absolute path',
-#      },
+      # use validate_absolute_path()
+      #      'absolute_path' => {
+      #        :name    => ['autofs_sysconfig','autofs_auto_master'],
+      #        :valid   => ['/absolute/filepath','/absolute/directory/'],
+      #        :invalid => ['invalid',3,2.42,['array'],a={'ha'=>'sh'}],
+      #        :message => 'is not an absolute path',
+      #      },
       'bool_stringified' => {
-        :name    => ['maps_hiera_merge','use_nis_maps','use_dash_hosts_for_net','service_enable'],
-        :valid   => [true,'true',false,'false'],
-        :invalid => [nil,'invalid',3,2.42,['array'],a={'ha'=>'sh'}],
-        :message => '(is not a boolean|Unknown type of boolean)',
+        name:    ['maps_hiera_merge', 'use_nis_maps', 'use_dash_hosts_for_net', 'service_enable'],
+        valid:   [true, 'true', false, 'false'],
+        invalid: [nil, 'invalid', 3, 2.42, ['array'], { 'ha' => 'sh' }],
+        message: '(is not a boolean|Unknown type of boolean)',
       },
-# use validate_hash()
-#      'hash' => {
-#        :name    => ['maps'],
-#        :valid   => [a={'ha'=>'sh'},a={'test1@test.void'=>'destination1','test2@test.void'=>['destination2','destination3']}],
-#        :invalid => [true,false,'invalid',3,2.42,['array']],
-#        :message => 'is not a Hash',
-#      },
-# use validate_integer(value,max,min)
-#      'integer' => {
-#        :name    => ['timeout','negative_timeout','mount_wait','umount_wait','mount_nfs_default_protocol'],
-#        :valid   => ['242','-242'],
-#        :invalid => ['invalid',2.42,['array'],a={'ha'=>'sh'}],
-#        :message => 'must be an integer',
-#      },
+      # use validate_hash()
```

Comment:
Any reason why we're keeping this code commented?

Reply:
These are waiting for the implementation of the given functionaliy. So they are a reminder so to say.

Reply:
OK.